src/NodeCompiler/CompileNodeToValue.php

```diff
@@ -15,51 +16,67 @@ class CompileNodeToValue
      */
     public function __invoke(Node $node)
     {
-        if ($node instanceof Node\Scalar\String_
-            || $node instanceof Node\Scalar\DNumber
-            || $node instanceof Node\Scalar\LNumber) {
-            return $node->value;
+        if (!$this->isCompilable($node)) {
+            throw new Exception\UnableToCompileNode('Unable to compile expression: ' . get_class($node));
         }
 
-        if ($node instanceof Node\Expr\Array_) {
-            $compiledArray = [];
-            foreach ($node->items as $arrayItem) {
-                $compiledValue = $this->__invoke($arrayItem->value);
-
-                if (null == $arrayItem->key) {
-                    $compiledArray[] = $compiledValue;
-                    continue;
-                }
+        if ($node instanceof Node\Expr\ConstFetch) {
+            return $this->compileConstFetch($node);
+        }
 
-                $compiledArray[$this->__invoke($arrayItem->key)] = $compiledValue;
-            }
-            return $compiledArray;
+        if ($node instanceof Node\Expr\ClassConstFetch) {
+            return $this->compileClassConstFetch($node);
         }
 
-        if ($node instanceof Node\Expr\ConstFetch) {
-            $firstName = reset($node->name->parts);
-            switch ($firstName) {
-                case 'null':
-                    return null;
-                case 'false':
-                    return false;
-                case 'true':
-                    return true;
-                default:
-                    // @todo this should evaluate the VALUE, not the name
-                    /* @see https://github.com/Roave/BetterReflection/issues/19 */
-                    return $firstName;
-            }
+        $printer = new Printer();
+        $code = $printer->prettyPrint([$node]);
+
+        eval('$x = ' . $code);
```

Comment:
Aye, ConstFetch and ClassConstFetch have dedicated handlers above (although I haven't dealt with them properly yet still). It's mainly my concern using `eval` tbh...